### PR TITLE
potential fix for 5932

### PR DESCRIPTION
**Pull request by dsyme**
Possible fix for perf problem #5932

### CODE REVIEW

**Comment by dsyme** [comment]
seems to work on my local machine

**Comment by TIHan** [comment]
So, CI passes on your local machine?

**Comment by dsyme** [comment]
OK, this is now ready.  I included some cleanup and some notes about the somewhat odd use of path normalization in that code (which I couldn't remove easily but I left some notes about how we might go about removing it)

File: src/fsharp/range.fs [hunk 1]
```diff
@@ -106,24 +125,64 @@ let _ = assert (endColumnMask =   mask64 endColumnShift   endColumnBitCount)
 let _ = assert (isSyntheticMask = mask64 isSyntheticShift isSyntheticBitCount)
 #endif
 
-// This is just a standard unique-index table
+/// Removes relative parts from any full paths
+let normalizeFilePath filePath =
```
Comment (cartermp) [inline src/fsharp/range.fs]:
```suggestion
let normalizeFilePath (filePath: string) = 
```

File: src/fsharp/range.fs [hunk 2]
```diff
@@ -106,24 +125,64 @@ let _ = assert (endColumnMask =   mask64 endColumnShift   endColumnBitCount)
 let _ = assert (isSyntheticMask = mask64 isSyntheticShift isSyntheticBitCount)
 #endif
 
-// This is just a standard unique-index table
+/// Removes relative parts from any full paths
+let normalizeFilePath filePath = 
+    try 
+        if Path.IsPathRooted filePath then
```
Comment (cartermp) [inline src/fsharp/range.fs]:
```suggestion
        if Path.IsPathRooted(filePath) then 
```

File: src/fsharp/range.fs [hunk 3]
```diff
@@ -106,24 +125,64 @@ let _ = assert (endColumnMask =   mask64 endColumnShift   endColumnBitCount)
 let _ = assert (isSyntheticMask = mask64 isSyntheticShift isSyntheticBitCount)
 #endif
 
-// This is just a standard unique-index table
+/// Removes relative parts from any full paths
+let normalizeFilePath filePath = 
+    try 
+        if Path.IsPathRooted filePath then 
+            Path.GetFullPath filePath
```
Comment (cartermp) [inline src/fsharp/range.fs]:
```suggestion
            Path.GetFullPath(filePath)
```

**Comment by dsyme** [comment]
@cartermp Done, thanks

**Pull request merged**